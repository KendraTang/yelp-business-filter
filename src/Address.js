import nl2br from 'react-nl2br'

export default (address) => 
  // Add whitespace between ASCII and CJK characters, and convert newline to
  // <br>
  nl2br(address
    .replace(/(\w)(?=[\u2E80-\u9FFF])/g, '$1 ')
    .replace(/([\u2E80-\u9FFF])(?=\w)/g, '$1 ')
  )
