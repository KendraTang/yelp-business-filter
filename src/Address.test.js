import Address from './Address'

it('Format address', () => {
  expect(Address('104 台北市中山區\n民族東路410巷2弄18號')).toMatchSnapshot()
})
