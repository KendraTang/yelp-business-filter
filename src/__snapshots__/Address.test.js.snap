// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Format address 1`] = `
Array [
  "104 台北市中山區",
  <br />,
  "民族東路 410 巷 2 弄 18 號",
]
`;
